testing/db-testkit/src/main/kotlin/net/corda/db/testkit/AbstractDBHelper.kt

```diff
@@ -0,0 +1,97 @@
+
+    abstract override fun getDatabase(): String
+
+    abstract override fun getAdminUser(): String
+
+    abstract override fun getAdminPassword(): String
+
+    abstract val port: String
+
+    abstract val host: String
+
+    abstract var jdbcUrl: String
+
+    abstract val driverClass: String
+
+    override fun getEntityManagerConfiguration(
+        inMemoryDbName: String,
+        dbUser: String?,
+        dbPassword: String?,
+        schemaName: String?,
+        createSchema: Boolean,
+        showSql: Boolean,
+        rewriteBatchedInserts: Boolean
+    ): EntityManagerConfiguration {
+        val ds = createDataSource(dbUser,dbPassword, schemaName, createSchema, rewriteBatchedInserts)
+        return DbEntityManagerConfiguration(ds,showSql,true, DdlManage.NONE)
+    }
+
+    override fun createDataSource(
+        dbUser: String?,
+        dbPassword: String?,
+        schemaName: String?,
+        createSchema: Boolean,
+        rewriteBatchedInserts: Boolean
+    ): CloseableDataSource {
+        val user = dbUser ?: getAdminUser()
+        val password = dbPassword ?: getAdminPassword()
+
+        if(!schemaName.isNullOrBlank()){
+            if(createSchema){
+                logger.info("Creating schema: $schemaName".emphasise())
+                createDataSource(driverClass,jdbcUrl,user,password, maximumPoolSize = 1)
+                    .connection.createSchema(schemaName)
+            }
+            val jdbcUrlCopy = if (rewriteBatchedInserts) {
+                "$jdbcUrl?currentSchema=$schemaName&reWriteBatchedInserts=true"
+            } else {
+                "$jdbcUrl?currentSchema=$schemaName"
+            }
+            logger.info("Using URL $jdbcUrlCopy".emphasise())
+            return createDataSource(driverClass,jdbcUrlCopy, user, password)
+        }
+        logger.info("Using URL $jdbcUrl".emphasise())
+        return createDataSource(driverClass,jdbcUrl, user, password)
+    }
+
+    override fun createConfig(
+        inMemoryDbName: String,
+        dbUser: String?,
+        dbPassword: String?,
+        schemaName: String?
+    ): Config {
+        val user = dbUser ?: getAdminUser()
+        val password = dbPassword ?: getAdminPassword()
+        if(!schemaName.isNullOrBlank()){
+            val jdbcUrlCopy = "$jdbcUrl?currentSchema=$schemaName"
+            return ConfigFactory.empty()
+                .withValue(DatabaseConfig.JDBC_URL, ConfigValueFactory.fromAnyRef(jdbcUrlCopy))
+                .withValue(DatabaseConfig.DB_USER, ConfigValueFactory.fromAnyRef(user))
+                .withValue(DatabaseConfig.DB_PASS, ConfigValueFactory.fromAnyRef(password))
+        }
+        return ConfigFactory.empty()
+            .withValue(DatabaseConfig.JDBC_URL, ConfigValueFactory.fromAnyRef(jdbcUrl))
+            .withValue(DatabaseConfig.DB_USER, ConfigValueFactory.fromAnyRef(user))
+            .withValue(DatabaseConfig.DB_PASS, ConfigValueFactory.fromAnyRef(password))
```

Comment:
Slighly more kotlin-y (and less code duplication)
```suggestion
        val currentJdbcUrl = if(!schemaName.isNullOrBlank()) "$jdbcUrl?currentSchema=$schemaName" else jdbcUrl
        return ConfigFactory.empty()
            .withValue(DatabaseConfig.JDBC_URL, ConfigValueFactory.fromAnyRef(currentJdbcUrl))
            .withValue(DatabaseConfig.DB_USER, ConfigValueFactory.fromAnyRef(user))
            .withValue(DatabaseConfig.DB_PASS, ConfigValueFactory.fromAnyRef(password))
```